Fixed OMPI for SUSE images

juhlee-microsoft opened the pull request:
New OMPI 4.0.2 requires below parameter.

  --mca btl_openib_allow_ib 1 --mca btl_openib_warn_no_device_params_found 0

waagent 2.2.45 is the only version for successful ib device initialization

iamshital commented on Utilities/CreateServicePrincipal.ps1:
```diff
@@ -0,0 +1,247 @@
+        New-AzureRmRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+    }
+    else {
+        New-AzureRmRoleAssignment -RoleDefinitionName "Virtual Machine Contributor" -ApplicationId $ClientId
```
Why not "Owner" for full privileges?

lizzha replied:
isn't Contributor enough for LISAv2 test?

juhlee-microsoft commented on Utilities/CreateServicePrincipal.ps1:
```diff
@@ -0,0 +1,246 @@
+    .FUNCTIONALITY
+        Generates random passwords
+    .LINK
+        http://blog.simonw.se/powershell-generating-random-password-for-active-directory/
```
Instead of it, can you post the one from TechNet? When you are in this web page and find a link called Technet Galley. Could be safe if we link Technet url than external blog url.

lizzha replied:
it was in the original code, I think I can remove this link since the TechNet link is on the top

juhlee-microsoft commented:
We wait for @lizzha update prior to merging.

lizzha commented:
@juhlee-microsoft  I've addressed the comments

juhlee-microsoft committed:
Updated OMPI param for new version

juhlee-microsoft committed:
1. Updated OMPI param for new version
2. Added the temp solution for waagent update
  a. set up the soft link for python in order to get the waagent version.
  b. enable ifconfig command
  c. restarted waagent before VM reboot
  d. restarted waagent after VM reboot in TestRDMA_MultiVM.sh
3. SUSE 12 SP4 has different pckg name
  a. net-tools-deprecated is only available in SLES 15 or above

juhlee-microsoft committed:
Added the temp solution for waagent update

juhlee-microsoft committed:
Fixed a few required steps

1. set up the soft link for python in order to get the waagent version.
2. enable ifconfig command
3. restarted waagent before VM reboot
4. restarted waagent after VM reboot in TestRDMA_MultiVM.sh

juhlee-microsoft committed:
Fixed the missing config update

juhlee-microsoft committed:
SUSE 12 SP4 has different pckg name

juhlee-microsoft committed:
net-tools-deprecated is only available in SLES 15 or above

juhlee-microsoft committed:
Update TestRDMA_MultiVM.sh

juhlee-microsoft committed:
Merge branch 'juhlee/opmi_suse_fix' of https://github.com/LIS/LISAv2 into juhlee/opmi_suse_fix

juhlee-microsoft committed:
Code sharing with RHEL8

Requires python3 before this function call.

LiliDeng reviewed (approved):
if tests pass, good to go
